### ISSUE TITLE
Getting "Unexpected Exception Loading Stream"

### ISSUE THREAD

**Issue by sapanda** [google/ExoPlayer#1333]
I get the errors below when attempting to read my 10mbps HEVC DASH stream. The servers are located (Belgium) far away from us (NYC) and the issue occurs after tons of buffering (Not sure if it would occur if the stream was smooth)

Content producing issue: http://146.148.19.157:1935/live/mc1_10.stream/manifest.mpd
Exoplayer Version: r1.5.4
Device: Samsung Galaxy Note 5 running Android 5.1.1
[dumpstate.txt](https://github.com/google/ExoPlayer/files/156268/dumpstate.txt)

```
E/LoadTask(19115): Unexpected exception loading stream
E/LoadTask(19115): java.lang.IllegalStateException: Top bit not zero: -32
E/LoadTask(19115): 	at com.google.android.exoplayer.util.ParsableByteArray.readUnsignedIntToInt(ParsableByteArray.java:240)
E/LoadTask(19115): 	at com.google.android.exoplayer.extractor.mp4.FragmentedMp4Extractor.parseTrun(FragmentedMp4Extractor.java:536)
E/LoadTask(19115): 	at com.google.android.exoplayer.extractor.mp4.FragmentedMp4Extractor.parseTraf(FragmentedMp4Extractor.java:361)
E/LoadTask(19115): 	at com.google.android.exoplayer.extractor.mp4.FragmentedMp4Extractor.parseMoof(FragmentedMp4Extractor.java:338)
E/LoadTask(19115): 	at com.google.android.exoplayer.extractor.mp4.FragmentedMp4Extractor.onMoofContainerAtomRead(FragmentedMp4Extractor.java:317)
E/LoadTask(19115): 	at com.google.android.exoplayer.extractor.mp4.FragmentedMp4Extractor.onContainerAtomRead(FragmentedMp4Extractor.java:282)
E/LoadTask(19115): 	at com.google.android.exoplayer.extractor.mp4.FragmentedMp4Extractor.readAtomPayload(FragmentedMp4Extractor.java:263)
E/LoadTask(19115): 	at com.google.android.exoplayer.extractor.mp4.FragmentedMp4Extractor.read(FragmentedMp4Extractor.java:170)
E/LoadTask(19115): 	at com.google.android.exoplayer.chunk.ChunkExtractorWrapper.read(ChunkExtractorWrapper.java:93)
E/LoadTask(19115): 	at com.google.android.exoplayer.chunk.ContainerMediaChunk.load(ContainerMediaChunk.java:165)
E/LoadTask(19115): 	at com.google.android.exoplayer.upstream.Loader$LoadTask.run(Loader.java:209)
E/LoadTask(19115): 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
E/LoadTask(19115): 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
E/LoadTask(19115): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
E/LoadTask(19115): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
E/LoadTask(19115): 	at java.lang.Thread.run(Thread.java:818)
E/EventLogger(19115): internalError [78.11, loadError]
E/EventLogger(19115): com.google.android.exoplayer.upstream.Loader$UnexpectedLoaderException: Unexpected IllegalStateException: Top bit not zero: -32
E/EventLogger(19115): 	at com.google.android.exoplayer.upstream.Loader$LoadTask.run(Loader.java:222)
E/EventLogger(19115): 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
E/EventLogger(19115): 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
E/EventLogger(19115): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
E/EventLogger(19115): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
E/EventLogger(19115): 	at java.lang.Thread.run(Thread.java:818)
E/EventLogger(19115): Caused by: java.lang.IllegalStateException: Top bit not zero: -32
E/EventLogger(19115): 	at com.google.android.exoplayer.util.ParsableByteArray.readUnsignedIntToInt(ParsableByteArray.java:240)
E/EventLogger(19115): 	at com.google.android.exoplayer.extractor.mp4.FragmentedMp4Extractor.parseTrun(FragmentedMp4Extractor.java:536)
E/EventLogger(19115): 	at com.google.android.exoplayer.extractor.mp4.FragmentedMp4Extractor.parseTraf(FragmentedMp4Extractor.java:361)
E/EventLogger(19115): 	at com.google.android.exoplayer.extractor.mp4.FragmentedMp4Extractor.parseMoof(FragmentedMp4Extractor.java:338)
E/EventLogger(19115): 	at com.google.android.exoplayer.extractor.mp4.FragmentedMp4Extractor.onMoofContainerAtomRead(FragmentedMp4Extractor.java:317)
E/EventLogger(19115): 	at com.google.android.exoplayer.extractor.mp4.FragmentedMp4Extractor.onContainerAtomRead(FragmentedMp4Extractor.java:282)
E/EventLogger(19115): 	at com.google.android.exoplayer.extractor.mp4.FragmentedMp4Extractor.readAtomPayload(FragmentedMp4Extractor.java:263)
E/EventLogger(19115): 	at com.google.android.exoplayer.extractor.mp4.FragmentedMp4Extractor.read(FragmentedMp4Extractor.java:170)
E/EventLogger(19115): 	at com.google.android.exoplayer.chunk.ChunkExtractorWrapper.read(ChunkExtractorWrapper.java:93)
E/EventLogger(19115): 	at com.google.android.exoplayer.chunk.ContainerMediaChunk.load(ContainerMediaChunk.java:165)
E/EventLogger(19115): 	at com.google.android.exoplayer.upstream.Loader$LoadTask.run(Loader.java:209)
E/EventLogger(19115): 	... 5 more
D/EventLogger(19115): bandwidth [78.32, 3812, 0.10, 6481084]
E/LoadTask(19115): Unexpected exception loading stream
E/LoadTask(19115): java.lang.IllegalStateException
E/LoadTask(19115): 	at com.google.android.exoplayer.util.Assertions.checkState(Assertions.java:66)
E/LoadTask(19115): 	at com.google.android.exoplayer.extractor.mp4.FragmentedMp4Extractor.readAtomHeader(FragmentedMp4Extractor.java:245)
E/LoadTask(19115): 	at com.google.android.exoplayer.extractor.mp4.FragmentedMp4Extractor.read(FragmentedMp4Extractor.java:165)
E/LoadTask(19115): 	at com.google.android.exoplayer.chunk.ChunkExtractorWrapper.read(ChunkExtractorWrapper.java:93)
E/LoadTask(19115): 	at com.google.android.exoplayer.chunk.ContainerMediaChunk.load(ContainerMediaChunk.java:165)
E/LoadTask(19115): 	at com.google.android.exoplayer.upstream.Loader$LoadTask.run(Loader.java:209)
E/LoadTask(19115): 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
E/LoadTask(19115): 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
E/LoadTask(19115): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
E/LoadTask(19115): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
E/LoadTask(19115): 	at java.lang.Thread.run(Thread.java:818)
E/EventLogger(19115): internalError [78.32, loadError]
E/EventLogger(19115): com.google.android.exoplayer.upstream.Loader$UnexpectedLoaderException: Unexpected IllegalStateException: null
E/EventLogger(19115): 	at com.google.android.exoplayer.upstream.Loader$LoadTask.run(Loader.java:222)
E/EventLogger(19115): 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
E/EventLogger(19115): 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
E/EventLogger(19115): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
E/EventLogger(19115): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
E/EventLogger(19115): 	at java.lang.Thread.run(Thread.java:818)
E/EventLogger(19115): Caused by: java.lang.IllegalStateException
E/EventLogger(19115): 	at com.google.android.exoplayer.util.Assertions.checkState(Assertions.java:66)
E/EventLogger(19115): 	at com.google.android.exoplayer.extractor.mp4.FragmentedMp4Extractor.readAtomHeader(FragmentedMp4Extractor.java:245)
E/EventLogger(19115): 	at com.google.android.exoplayer.extractor.mp4.FragmentedMp4Extractor.read(FragmentedMp4Extractor.java:165)
E/EventLogger(19115): 	at com.google.android.exoplayer.chunk.ChunkExtractorWrapper.read(ChunkExtractorWrapper.java:93)
E/EventLogger(19115): 	at com.google.android.exoplayer.chunk.ContainerMediaChunk.load(ContainerMediaChunk.java:165)
E/EventLogger(19115): 	at com.google.android.exoplayer.upstream.Loader$LoadTask.run(Loader.java:209)
E/EventLogger(19115): 	... 5 more
```

**Comment by ojw28** [comment]
The content link gives a 404 when I try and access it. Is it no longer available and/or is it geo-restricted?

**Comment by sapanda** [comment]
Sorry about that, server was down. It's back up and running.

**Comment by ojw28** [comment]
Thanks. One point of clarification - I assume it's just a side effect of the way you've produced the test content that the manifest is marked as being dynamic even though it doesn't actually update? It causes the player to get stuck in a never-ending buffering state as it waits for more segments to be added to the manifest. If so then fine; just thought I'd check!

**Comment by ojw28** [comment]
I tried quite hard to "break this" and couldn't reproduce with `1.5.6`, even when I inserted a whole bunch of artificial network failures/delays etc. 

One source of this kind of issue can be if your server isn't handling range requests properly (e.g. we load part of a segment, a network failure occurs, we make a range request to load just the remainder of the segment, the server serves us incorrect data, we fail). I can't see any evidence of that happening though; your server seems to respond correctly.

I don't think much changed between `1.5.4` and `1.5.6`, but could you try again using the latest ExoPlayer demo app + your URL, and see if you can still reproduce it there?

**Comment by sapanda** [comment]
> I assume it's just a side effect of the way you've produced the test content that the manifest is marked as being dynamic even though it doesn't actually update.

I should probably have provided additional info - the DASH stream is being generated by our Wowza Streaming Engine instance. I'm using their stream scheduling feature to simulate a live stream which is on loop. That probably explains the "dynamic" type?

I'll give 1.5.6 a shot and let you know if the issue still occurs. Thanks!

**Comment by ojw28** [comment]
Not sure if this is still an issue, but closing since we weren't able to reproduce it and no further information was provided.

**Comment by byronclark** [comment]
Having hit the same issue with Wowza, I can confirm that it was fixed in Wowza 4.4.0 and then broken again in Wowza 4.4.1.

**Comment by ranithsachin** [comment]
Wowza 4.5.0 has this issue solved, but this issue is present in Wowza 4.4.1.

**Comment by AquilesCanta** [comment]
@byronclark , @ranithsachin : If you really want to figure out what is going on with 4.4.1, I'd suggest you get a copy of the manifest and mp4 segments that generate the error and provide us with a link (feel free to share this info by emailing dev.exoplayer@gmail.com). We need to be deterministically able to reproduce the issue. With no more information, everything points to the media generation. Also, please file a new issue with the required information.
